network/benchmark/src/lib.rs

```diff
@@ -295,6 +279,64 @@ pub async fn run_netbench_service(
     }
 }
 
+async fn connection_listener(
+    node_config: NodeConfig,
+    network_client: NetworkClient<NetbenchMessage>,
+    time_service: TimeService,
+    shared: Arc<RwLock<NetbenchSharedState>>,
+    handle: Handle,
+) {
+    let config = node_config.netbench.unwrap();
+    let peers_and_metadata = network_client.get_peers_and_metadata();
+    let mut connected_peers = HashSet::new();
+    let mut connection_notifications = peers_and_metadata.subscribe();
+    loop {
+        match connection_notifications.recv().await {
+            None => {
+                info!("netbench connection_listener exit");
+                return;
+            },
+            Some(note) => match note {
+                ConnectionNotification::NewPeer(meta, network_id) => {
+                    let peer_network_id = PeerNetworkId::new(network_id, meta.remote_peer_id);
+                    if connected_peers.contains(&peer_network_id) {
+                        continue;
```

Comment:
nit: I'd maybe log an error as this shouldn't really happen?

Reply:
it was designed to handle the previous mode of doing a fetch and then starting the subscription and being idempotent, so it's harmless if it gets a NewPeer for a peer it already knows about.